src/init.c

```diff
@@ -618,7 +618,28 @@ static int hyper_kill_container(char *json, int length)
 		goto out;
 	}
 
-	kill(c->exec.pid, (int)json_object_get_number(json_object(value), "signal"));
+	const int all_processes = json_object_get_boolean(json_object(value),
+		"allProcesses");
+
+	const int signal = (int)json_object_get_number(json_object(value), "signal");
+
+	struct hyper_exec *exec;
+	/* json_object_get_boolean returns '1' only when
+	 * 'allProcesses' exists and is true
+	 */
+	if (1 == all_processes) {
```

Comment:
for security reasons I'd prefer to use ```if (1 == all_processes)```
because if I make a mistake like this ```if (1 = all_processes)``` compiler will fail
otherwise with ```if (all_processes = 1)``` condition will be always true